### PR TITLE
Improve ServiceAccount administration doc

### PR DESCRIPTION
**Pull request by tengqm**
This PR fixes some nits in the doc and slightly revised the content to
conform to content guidelines.

### CODE REVIEW

**Comment by kbhawkey** [comment]
👀

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-23842--kubernetes-io-master-staging.netlify.app/docs/reference/access-authn-authz/service-accounts-admin/

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 1]
```diff
@@ -23,96 +23,108 @@ incomplete features are referred to in order to better describe service accounts
...
-    ad-hoc and have namespaced names, such config is portable.
+- User accounts are for humans. Service accounts are for processes, which run
+  in pods.
+- User accounts are intended to be global. Names must be unique across all
+  namespaces of a cluster. Service accounts are namespaced.
+- Typically, a cluster's user accounts might be synced from a corporate
+  database, where new user account creation requires special privileges and is
+  tied to complex business processes. Service account creation is intended to be
+  more lightweight, allowing cluster users to create service accounts for
+  specific tasks (i.e. principle of least privilege).
```
Comment (kbhawkey) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:34]: nit: replace "i.e." with "for example" or "such as"
nit: replace "ad-hoc" with "ad hoc" or "automatically"?

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 2]
```diff
@@ -23,96 +23,108 @@ incomplete features are referred to in order to better describe service accounts
...
 ## Service account automation
 
 Three separate components cooperate to implement the automation around service accounts:
 
-  - A Service account admission controller
-  - A Token controller
-  - A Service account controller
+- A `ServiceAccount` admission controller
+- A Token controller
+- A ServiceAccount controller
```
Comment (kbhawkey) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:46]: nit: Are you formatting lines 44 and 46?

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 3]
```diff
@@ -23,96 +23,108 @@ incomplete features are referred to in order to better describe service accounts
...
+- watches ServiceAccount token Secret addition, and ensures the referenced
+  ServiceAccount exists, and adds a token to the Secret if needed.
+- watches Secret deletion and removes a reference from the corresponding
+  ServiceAccount if needed.
 
-You must pass a service account private key file to the token controller in the controller-manager by using
-the `--service-account-private-key-file` option. The private key will be used to sign generated service account tokens.
-Similarly, you must pass the corresponding public key to the kube-apiserver using the `--service-account-key-file`
-option. The public key will be used to verify the tokens during authentication.
+You must pass a service account private key file to the token controller in
```
Comment (kbhawkey) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:88]: "token controller" vs TokenController?

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 4]
```diff
@@ -23,96 +23,108 @@ incomplete features are referred to in order to better describe service accounts
...
 ```
 
 ```shell
-kubectl create -f ./secret.json
+kubectl create -f ./secret.yaml
 kubectl describe secret mysecretname
 ```
 
-#### To delete/invalidate a service account token
+#### To delete/invalidate a ServiceAccount token Secret
```
Comment (kbhawkey) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md:120]: secret should be capitalized?

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md:34]: ok

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 2 above)
Comment (tengqm) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md:46]: Yes.

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 3 above)
Comment (tengqm) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md:88]: I thought about that. If we put this as TokenController, it means we have something named TokenController. However, it is not the case. Even if we do have such struct in GoLang, we don't want to expose it to readers. This is the reason I didn't change 'token controller' to 'TokenController'.

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 4 above)
Comment (tengqm) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md:120]: yes, we are referring to a Secret object, rather than a general term.

**Comment by kbhawkey** [comment]
Thanks @tengqm .
/lgtm

**Pull request merged**